corelib-web/src/main/java/eu/europeana/corelib/web/exception/ProblemType.java

```diff
@@ -31,8 +31,11 @@
 
     // Other errors
     // ------------
-    APIKEY_INVALID(null, "Invalid API key", ProblemResponseAction.IGNORE),
-    APIKEY_MISSING(null, "No API key provided", ProblemResponseAction.IGNORE),
+    APIKEY_INVALID("410-AI", "Invalid API key", ProblemResponseAction.IGNORE),
+    APIKEY_MISSING("400-AM", "No API key provided", ProblemResponseAction.IGNORE),
+    APIKEY_NOT_REGISTERED("401-AR", "API key is not registered", ProblemResponseAction.IGNORE),
```

Comment:
I think APIKEY_INVALID is what we use now when an API key does not exist, so it's the same as APIKEY_NOT REGISTERED. I propose to remove the latter (because I like the name APIKEY_INVALID) better.

Reply:
We don't use APIKEY_INVALID if an API key does not exist. If that's the case, the Apikey service responds with HTTP 401, and the Search API with APIKEY_NOT_REGISTERED:

> {
>   "apikey": "ortlansib",
>   "success": false,
>   "error": "API key is not registered",
>   "code": "401-AR"
> }

... which is correct: it is "not registered". 

"Invalid" implies something else: it exists but it isn't valid. 
In that case, the Apikey service responds with HTTP 410, and the Api translates that like this: 

> {
>   "apikey": "ortlansia",
>   "success": false,
>   "error": "API key is deprecated",
>   "code": "410-AD"
> }

... which is IMO the correct response to the situation.

Reply:
Maybe you meant APIKEY_DEPRECATED instead of APIKEY_NOT_REGISTERED as competing with APIKEY_INVALID?
That's indeed a doubling, they both represent a HTTP 410 error. We could change the 410 error to use APIKEY_INVALID, indeed.

Let's decide tomorrow morning.

Reply:
after [discussing](https://europeana.atlassian.net/browse/EA-1826) we settled on:
HTTP 401: when there is no record for that key (never existed)
HTTP 410: when the api exists but was deprecated in the meantime
HTTP 200 is implicit when the the apikey is valid.

Additionally, a HTTP 400 is given if no API key is supplied, similar to how the Apikey service responds to a validation request when no API key is presented)

Re. the error messages:
HTTP 401 -> APIKEY_DOES_NOT_EXIST "API key doesn't exist" (replacing the ambiguous APIKEY_INVALID)
HTTP 410 -> APIKEY_DEPRECATED "API key is deprecated"
HTTP 400 -> APIKEY_MISSING "No API key provided"

If there is a connectivity issue (either to the database for old-school requests, or the apikey service) we do not return any error and suspend validation. We do log an error in such cases. 

The Limitresponse is now deprecated, we do not return the "999" any longer; and we only create an APIkey object in the old school validation method; no temp APIkey objects are being generated any longer.